File: web/components/accordion/accordion-demo.dart
import "package:angular2/angular2.dart";
import 'package:ng2-strap/accordion/accordion.dart';

@Component(selector: "accordion-demo")
@View(
    templateUrl: "./accordion-demo.html",
    directives: const [ACCORDION_DIRECTIVES, NgClass, CORE_DIRECTIVES, FORM_DIRECTIVES])
class AccordionDemo {
  bool oneAtATime = true;

  List<String> items = ["Item 1", "Item 2", "Item 3"];

  Map status = {"isFirstOpen": true, "isFirstDisabled": false, 'open': false};

  bool isOpen = false;

  List<dynamic> groups = [
    {"title": "Dynamic Group Header - 1", "content": "Dynamic Group Body - 1"},
    {"title": "Dynamic Group Header - 2", "content": "Dynamic Group Body - 2"}
  ];

  addItem() {
    this.items.add('Item ${this.items.length + 1}');
  }
}

File: demo/components/accordion/accordion-demo.ts
import { Component } from '@angular/core';

// webpack html imports
let template = require('./accordion-demo.html');

@Component({
  selector: 'accordion-demo',
  template: template
})
export class AccordionDemoComponent {
  public oneAtATime:boolean = true;
  public items:Array<string> = ['Item 1', 'Item 2', 'Item 3'];

  public status:Object = {
    isFirstOpen: true,
    isFirstDisabled: false
  };

  public groups:Array<any> = [
    {
      title: 'Dynamic Group Header - 1',
      content: 'Dynamic Group Body - 1'
    },
    {
      title: 'Dynamic Group Header - 2',
      content: 'Dynamic Group Body - 2'
    }
  ];

  public addItem():void {
    this.items.push(`Items ${this.items.length + 1}`);
  }
}
